OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -1055,15 +1052,13 @@ static bool Faction(S server, Connection conn, Session.Client client, string s)
 				if (server.LobbyInfo.Slots[targetClient.Slot].LockFaction)
 					return true;
 
-				var factions = server.Map.WorldActorInfo.TraitInfos<FactionInfo>()
-					.Where(f => f.Selectable).Select(f => f.InternalName)
-					.ToList();
-
 				var faction = parts[1];
-				if (!factions.Contains(faction))
+				var isValidFaction = server.Map.WorldActorInfo.TraitInfos<FactionInfo>()
+					.Any(f => f.Selectable && f.InternalName == client.Faction);
+
+				if (!isValidFaction)
 				{
 					server.SendFluentMessageTo(conn, InvalidFactionSelected, new object[] { "faction", faction });
-					server.SendFluentMessageTo(conn, SupportedFactions, new object[] { "factions", factions.JoinWith(", ") });
```

Comment:
We don't report these for any of the other options, and this info isn't very useful for the client - so better removed.